core/consensus/src/adapter.rs

```diff
@@ -285,6 +288,60 @@ where
             .await?;
         Ok(res.inner)
     }
+
+    /// Pull a proof of certain block from other nodes
+    async fn get_proof_from_remote(
```

Comment:
The caller can directly specify which height of proof to get.

When the receiver receives the message, it can choose to return proof from storage or memory  on its current height.

Reply:
I did add the logic you mentioned on storage and endpoint handler. but then I changed my code and add this to put all logic to sender because I don't want to add more complicated into storage because storage is no more than a key-value datacenter and providor of simple get/set functions. alse adding this apart of code into message handler will put on overhead to the receiver, but I thinks it should goes to the sender.  One in one, sender handle the logic of determine, not receiver

Reply:
changed the logic , now the caller can chose which height she need